### PR TITLE
[#18] Add unit converter

### PR DESCRIPTION
**Pull request by todofixthis**

### CODE REVIEW

File: test/api_test.py [hunk 1]
```diff
@@ -135,3 +135,14 @@ def test_create_command(self):
...
+  def test_convert_units(self):
+    """
+    Converting between iota standard units.
+    """
+    api = Iota(MockAdapter())
+
+    self.assertEqual(api.convert_units('1.618 Mi', 'i'), 1618000)
+    self.assertEqual(api.convert_units('42 i', 'Ki'), 0.042)
+    with self.assertRaises(ValueError):
+      api.convert_units('3.141596 Xi', 'Pi')
```
Comment (todofixthis) [inline test/api_test.py]: `convert_value_to_standard_unit` deserves its own test case.  This makes it easier to test just that function in isolation from the rest of the library.

Also, I would like each scenario split into its own test (e.g., create `test_convert_to_smaller_unit`, `test_convert_to_bigger_unit`, `test_error_invalid_unit`, and so on).  This makes it easier to understand what is being tested, and if any assertion fails, it doesn't prevent the other assertions from being checked.

Lastly, there are a few more tests that need to be added for full coverage:

- Input is the wrong type (e.g., `[1.618, 'Mi']`).
- Input is already a float value (e.g., `1.618`).  Probably same result as the previous one.
- Input is empty (i.e., `''`).
- Input has too few parts (e.g., `'1.618`).
- Input is missing space (e.g., `'1.618Mi'`).  We can treat this as an error.
- Input has too many parts (e.g., `'1 618 Mi'`).
- Result includes fractional IOTAS (e.g., `'1.6189 Ki'`).  This is not an error, just using unit tests to document that this behavior is allowed.
- Input is negative (e.g., `'-1.618 Mi'`).
- Input uses different case (e.g., `'1.618 mI'`).  This should probably be an error... probably.

File: src/iota/transaction.py [hunk 2]
```diff
@@ -45,6 +46,49 @@ def get_current_timestamp():
...
+
+    :return:
+      float as units of symbol to convert to
+  """
+  value_tuple = value.split()
+
+  try:
+    # Get input value
+    amount = float(value_tuple[0])
+  except (ValueError, IndexError) as e:
```
Comment (todofixthis) [inline src/iota/transaction.py]: Is `e` being used here?  Double-check your IDE's configuration; it should highlight this as an issue.

File: src/iota/__init__.py [hunk 3]
```diff
@@ -17,6 +17,17 @@
 In that way, it's kind of like toxic waste in a superhero story.
 """
 
+STANDARD_UNITS = {
+    # Valid IOTA unit prefixes. Example value '25.2 Mi'
```
Comment (todofixthis) [inline src/iota/__init__.py]: ... and *that's* the first thing that appears in this PR.  Great; put my pedantry on display front and center :P

File: test/api_test.py (hunk 1 above)
Comment (curVV) [inline test/api_test.py]: I've created a test case `class UnitConverterTestCase(TestCase)` in `test/transaction_test.py`
How do run only this test case?
Can it be done using setup.py?

Comment (curVV) [inline test/api_test.py]: OK, I've managed to at least run all tests in `transaction.py` with `python setup.py test -s test.transaction_test`
Not getting it to run `test.transaction_test.UnitConverterTestCase` only, though. Unit test runs but reports: `Ran 0 tests in 0.001s`

**Comment by todofixthis** [comment]
Changes look good!  Thanks!

**Pull request merged**